config/routes.rb

```diff
@@ -1,14 +1,21 @@
 MyLibraryNYC::Application.routes.draw do
- 
-  devise_for :users, :path => "users", :path_names => { :sign_in => 'start', :sign_out => 'signout', :sign_up => 'signup' }, :controllers => { :registrations => :registrations }
-  
+
+  devise_for :users, :path => "users", :path_names => { :sign_in => 'start', :sign_out => 'signout', :sign_up => 'signup' }, :controllers => { :registrations => :registrations, :sessions => :sessions }
+
+  devise_scope :user do
+    match 'timeout_check' => 'sessions#timeout_check', via: :get
+    match 'timeout' => 'sessions#timeout', via: :get
+  end
+
+  get 'extend_session_iframe' => 'home#extend_session_iframe'
```

Comment:
Maybe add some comments on this route? A future developer might be caught off guard by this

Reply:
I added a comment in the controller rather than the routes.rb.